src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,58 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int AMOUNT_INDEX = 1;
+    private static final int TYPE_INDEX = 0;
+    private static final String SUPPLY = "supply";
+    private static final String RESULT = "result";
+    private static final String BUY = "buy";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String text = readFile(fromFileName);
+        writeFile(toFileName, text);
+    }
+
+    private String readFile(String fromFile) {
```

Comment:
This method is doing more than just reading a file. It's also calculating the statistics and building the report. Consider separating these responsibilities into different methods. This is related to the 'Divide program logic' item from the checklist.